Add initial documentation of deploy-contract

ThunderDeliverer opened the pull request:
The `deploy-contract` example has been documented in order to explain how to build it.

TODO:

- [x] Verify the flow of the documentation
- [x] Add line breaks at column 100

ThunderDeliverer committed:
Add initial documentation of deploy-contract

The deploy-contract example has been documented in order to explain how
to build it.

shunjizhan commented on deploy-contract/README.md:
```diff
@@ -1,63 +1,829 @@
+# Bodhi.js example: Contract Interaction
 
-app: https://bodhi-example-contract.vercel.app/
+A basic example on how to sign Acala EVM+ transactions with a Polkadot vallet and the [bodhi.js](https://github.com/AcalaNetwork/bodhi.js/) dependency.
```
```suggestion
A basic example on how to sign Acala EVM+ transactions with a Polkadot wallet and the [bodhi.js](https://github.com/AcalaNetwork/bodhi.js/) sdk.
```

shunjizhan commented on deploy-contract/README.md:
`````diff
@@ -1,63 +1,829 @@
+
+Once the setup is complete, add the build target to the [`vite.config.ts`](./vite.config.ts):
+
+````typescript
`````
isn't this part auto generated?

ThunderDeliverer replied:
`vite.config.ts` is autogenerated, but the build target had to be added. It was in your example, but it wasn't in my parallel one. Once I added the build target to mine, `yarn build` finally managed to build the project

shunjizhan commented on deploy-contract/README.md:
`````diff
@@ -1,63 +1,829 @@
+const Check = () => (<span className='check'>✓</span>);
+````
+
+We can start populating the `App` function with type definitions. We have to define the extensions, status flags and data:
`````
```suggestion
We can start populating the `App` states by defining state hooks for extensions, status flags and data :
```

shunjizhan commented on deploy-contract/README.md:
`````diff
@@ -1,63 +1,829 @@
+
+In addition to the functionalities that we added above, we also require the ability for users to select the wallet extension that they want to use with the dApp and the ability for them to select an account from that extension:
+
+````typescript
`````
```suggestion
````tsx
```

shunjizhan commented on deploy-contract/README.md:
`````diff
@@ -1,63 +1,829 @@
+
+All of the functionality of our dApp is ready to be presented in the interface. We will wrap the interface in a `return` statement as we want our `App` function to return it:
+
+````typescript
`````
```suggestion
````tsx
```

shunjizhan commented on deploy-contract/README.md:
`````diff
@@ -1,63 +1,829 @@
+
+In the first section of the interface, we will utilize the chain node connection mechanic:
+
+````typescript
`````
```suggestion
````tsx
```

shunjizhan commented on deploy-contract/README.md:
`````diff
@@ -1,63 +1,829 @@
+
+In the second section we will use the mechainc to connect to the wallet extensions of the user's browser and use the utilities to select the preferred wallet extension and preferred account for the interaction with the chain. It will also provide the information about the connected account:
+
+````typescript
`````
```suggestion
````tsx
```

shunjizhan reviewed (approved):
Thanks Jan! Looks great, just had a couple small modification suggestions.

Also, let's change all react related code from `ts` to `tsx` for better syntax highlighting

ThunderDeliverer committed:
Minor rewording and cosmetic updates

ThunderDeliverer committed:
Added a missing line of App.tsx

ThunderDeliverer commented:
@shunjizhan would you mind quickly going through the tutorial and copy pasting everything in a tmp project to see if you are able to run `yarn dev`? It's throwing errors for me, but it seems like it shouldn't 🤔

ThunderDeliverer committed:
Final touches to initial version of deploy-contract

Added image preview of the dApp and minor content refactor to get the
example through the finish line.

ThunderDeliverer committed:
Add line breaks to enforce the 100 character limit
